graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/DgsExecutionResult.kt

```diff
@@ -133,4 +86,56 @@ class DgsExecutionResult @JvmOverloads constructor(
 
         return spec
     }
+
+    // Refer to https://github.com/Netflix/dgs-framework/pull/1261 for further details.
+    private fun addExtensionsHeaderKeyToHeader() {
+        if (executionResult.extensions?.containsKey(DGS_RESPONSE_HEADERS_KEY) == true) {
+            val dgsResponseHeaders = executionResult.extensions[DGS_RESPONSE_HEADERS_KEY]
+
+            if (dgsResponseHeaders is Map<*, *>) {
+                dgsResponseHeaders.forEach {
+                    if (it.key != null) {
+                        headers.add(it.key.toString(), it.value?.toString())
+                    }
+                }
+            } else {
+                logger.warn(
+                    "{} must be of type java.util.Map, but was {}",
+                    DGS_RESPONSE_HEADERS_KEY,
+                    dgsResponseHeaders?.javaClass?.name
+                )
+            }
+        }
+    }
+
+    /**
+     * Facilitate the construction of a [DgsExecutionResult] instance.
+     */
+    class Builder {
+        var executionResult: ExecutionResult = DEFAULT_EXECUTION_RESULT
+            private set
+        fun executionResult(executionResult: ExecutionResult) = apply { this.executionResult = executionResult }
+
+        fun executionResult(executionResultBuilder: ExecutionResultImpl.Builder) = apply { this.executionResult = executionResultBuilder.build() }
+
+        var headers: HttpHeaders = HttpHeaders()
```

Comment:
Can this default to `HttpHeaders.EMPTY`, so it doesn't need to instantiate anything if the builder method isn't called?

Reply:
It should, thanks @kilink !